src/SFA.DAS.AssessorService.Application/Handlers/UserManagement/RequestForPrivilegeHandler.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Api.Types.Models.UserManagement;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Consts;
+using SFA.DAS.AssessorService.Domain.Entities;
+using SFA.DAS.AssessorService.Settings;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.UserManagement
+{
+    public class RequestForPrivilegeHandler : IRequestHandler<RequestForPrivilegeRequest>
+    {
+        private readonly IMediator _mediator;
+        private readonly IContactQueryRepository _contactQueryRepository;
+        private readonly IOrganisationQueryRepository _organisationQueryRepository;
+        private readonly IWebConfiguration _config;
+        private readonly IContactRepository _contactRepository;
+
+        public RequestForPrivilegeHandler(IMediator mediator, IContactQueryRepository contactQueryRepository, 
+            IOrganisationQueryRepository organisationQueryRepository, IWebConfiguration config, IContactRepository contactRepository)
+        {
+            _mediator = mediator;
+            _contactQueryRepository = contactQueryRepository;
+            _organisationQueryRepository = organisationQueryRepository;
+            _config = config;
+            _contactRepository = contactRepository;
+        }
+        
+        public async Task Handle(RequestForPrivilegeRequest message, CancellationToken cancellationToken)
+        {
```

Comment:
Do you need to worry if anything returns null (i.e. it cannot find the contact or the privilege for example) ?

Reply:
Not in this case as we are already sure the contact and privilege exist as the Contact is the one trying to access the page (so must exist) and the privilege is what's stopped the contact accessing the page (so must exist). 😁